### ISSUE TITLE
Typings

### ISSUE THREAD

**Issue by Dqrshan** [shadowplay1/discord-economy-super#22]
The typings are messed up and throws a lot of errors.

```typescript
node_modules/discord-economy-super/mongodb/typings/cached/CachedItems.d.ts:20:40 - error TS7006: Parameter 'guildID' implicitly has an 'any' type.

20     public constructor(userID: string, guildID, options: EconomyOptions, database: DatabaseManager)
                                          ~~~~~~~

node_modules/discord-economy-super/mongodb/typings/cached/CachedItems.d.ts:23:49 - error TS2344: Type 'CooldownItem' does not satisfy the constraint 'EconomyConstructors'.
  Type 'CooldownItem' is missing the following properties from type 'HistoryItem<any>': id, name, price, totalPrice, and 7 more.

23 export class CachedCooldowns extends CachedItem<CooldownItem, true, false, false> {
                                                   ~~~~~~~~~~~~

node_modules/discord-economy-super/mongodb/typings/cached/CachedItems.d.ts:24:40 - error TS7006: Parameter 'guildID' implicitly has an 'any' type.

24     public constructor(userID: string, guildID, options: EconomyOptions, database: DatabaseManager)
                                          ~~~~~~~

node_modules/discord-economy-super/mongodb/typings/classes/CooldownItem.d.ts:21:21 - error TS2304: Cannot find name 'EconomyOptions'.

21         ecoOptions: EconomyOptions,
                       ~~~~~~~~~~~~~~

node_modules/discord-economy-super/mongodb/typings/classes/CooldownItem.d.ts:22:26 - error TS2304: Cannot find name 'RawEconomyUser'.

22         cooldownsObject: RawEconomyUser,
                            ~~~~~~~~~~~~~~

node_modules/discord-economy-super/mongodb/typings/classes/InventoryItem.d.ts:37:12 - error TS7008: Member 'guildID' implicitly has an 'any' type.

37     public guildID
              ~~~~~~~

node_modules/discord-economy-super/mongodb/typings/classes/user/Cooldowns.d.ts:1:10 - error TS2616: 'RewardCooldownData' can only be imported by using 'import RewardCooldownData = require("../../interfaces/RewardCooldownData")' or a default import.

1 import { RewardCooldownData } from '../../interfaces/RewardCooldownData'
           ~~~~~~~~~~~~~~~~~~

node_modules/discord-economy-super/mongodb/typings/classes/user/Cooldowns.d.ts:1:36 - error TS2497: This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export.

1 import { RewardCooldownData } from '../../interfaces/RewardCooldownData'
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/discord-economy-super/mongodb/typings/classes/user/History.d.ts:59:58 - error TS2304: Cannot find name 'SVGAnimatedNumberList'.

59     public findItem<T extends object = any>(id: string | SVGAnimatedNumberList): Promise<HistoryItem<T>>
                                                            ~~~~~~~~~~~~~~~~~~~~~

node_modules/discord-economy-super/mongodb/typings/classes/user/History.d.ts:66:57 - error TS2304: Cannot find name 'SVGAnimatedNumberList'.

66     public getItem<T extends object = any>(id: string | SVGAnimatedNumberList): Promise<HistoryItem<T>>
                                                           ~~~~~~~~~~~~~~~~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:11:5 - error TS1070: 'public' modifier cannot appear on a type member.

11     public id: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:16:5 - error TS1070: 'public' modifier cannot appear on a type member.

16     public memberID: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:21:5 - error TS1070: 'public' modifier cannot appear on a type member.

21     public guildID: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:26:5 - error TS1070: 'public' modifier cannot appear on a type member.

26     public name: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:31:5 - error TS1070: 'public' modifier cannot appear on a type member.

31     public price: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:36:5 - error TS1070: 'public' modifier cannot appear on a type member.

36     public totalPrice: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:41:5 - error TS1070: 'public' modifier cannot appear on a type member.

41     public quantity: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:46:5 - error TS1070: 'public' modifier cannot appear on a type member.

46     public message?: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:51:5 - error TS1070: 'public' modifier cannot appear on a type member.

51     public role?: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:56:5 - error TS1070: 'public' modifier cannot appear on a type member.

56     public date: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/HistoryData.d.ts:61:5 - error TS1070: 'public' modifier cannot appear on a type member.

61     public custom?: CustomItemData<T>
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/InventoryData.d.ts:11:5 - error TS1070: 'public' modifier cannot appear on a type member.

11     public id: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/InventoryData.d.ts:16:5 - error TS1070: 'public' modifier cannot appear on a type member.

16     public name: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/InventoryData.d.ts:21:5 - error TS1070: 'public' modifier cannot appear on a type member.

21     public price: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/InventoryData.d.ts:26:5 - error TS1070: 'public' modifier cannot appear on a type member.

26     public message?: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/InventoryData.d.ts:31:5 - error TS1070: 'public' modifier cannot appear on a type member.

31     public maxAmount?: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/InventoryData.d.ts:36:5 - error TS1070: 'public' modifier cannot appear on a type member.

36     public role?: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/InventoryData.d.ts:41:5 - error TS1070: 'public' modifier cannot appear on a type member.

41     public date: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/InventoryData.d.ts:46:5 - error TS1070: 'public' modifier cannot appear on a type member.

46     public custom?: CustomItemData<T>
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/ItemData.d.ts:11:5 - error TS1070: 'public' modifier cannot appear on a type member.

11     public id: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/ItemData.d.ts:16:5 - error TS1070: 'public' modifier cannot appear on a type member.

16     public name: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/ItemData.d.ts:21:5 - error TS1070: 'public' modifier cannot appear on a type member.

21     public price: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/ItemData.d.ts:26:5 - error TS1070: 'public' modifier cannot appear on a type member.

26     public message: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/ItemData.d.ts:31:5 - error TS1070: 'public' modifier cannot appear on a type member.

31     public description: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/ItemData.d.ts:36:5 - error TS1070: 'public' modifier cannot appear on a type member.

36     public maxAmount: number
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/ItemData.d.ts:41:5 - error TS1070: 'public' modifier cannot appear on a type member.

41     public role: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/ItemData.d.ts:46:5 - error TS1070: 'public' modifier cannot appear on a type member.

46     public date: string
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/ItemData.d.ts:51:5 - error TS1070: 'public' modifier cannot appear on a type member.

51     public custom: CustomItemData<T>
       ~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/RewardCooldownData.d.ts:8:11 - error TS2709: Cannot use namespace 'TimeObject' as a type.

8     time: TimeObject
            ~~~~~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/RewardObject.d.ts:3:10 - error TS2616: 'RewardCooldownData' can only be imported by using 'import RewardCooldownData = require("./RewardCooldownData")' or a default import.

3 import { RewardCooldownData } from './RewardCooldownData'
           ~~~~~~~~~~~~~~~~~~

node_modules/discord-economy-super/mongodb/typings/interfaces/RewardObject.d.ts:3:36 - error TS2497: This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export.

3 import { RewardCooldownData } from './RewardCooldownData'
```

**Comment by shadowplay1** [comment]
Can you give me the code that causes this bunch of errors? I'm already working on a new version with some new features and bugfixes and it would be very helpful at the moment. It's still unfinished, so the update and fixes thing will be out in tommorow's dev version

**Comment by shadowplay1** [comment]
let me know if it works without errors now or there's another bugs or errors

**Comment by shadowplay1** [comment]
fixed in `v1.6.2` 🚀!
